pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -199,6 +199,34 @@ curl "${ARGS[@]}" "$VAULT_ADDR"/"$VAULT_KV_VERS"/"$VAULT_BACKEND_PATH"/"$KEK_NAM
 
 # Put the KEK in a file for cryptsetup to read
 python3 -c "import sys, json; print(json.load(sys.stdin)[\"data\"][\"$KEK_NAME\"], end='')" < "$CURL_PAYLOAD" > "$KEY_PATH"
+`
+	blockDevMapper = `
+set -xe
+
+PVC_SOURCE=%s
+PVC_DEST=%s
+CP_ARGS=(--archive --dereference --verbose)
+
+if [ -b "$PVC_DEST" ]; then
+	PVC_SOURCE_MAJ_MIN=$(stat -c '%%t%%T' $PVC_SOURCE)
+	PVC_DEST_MAJ_MIN=$(stat -c '%%t%%T' $PVC_DEST)
+	if [ "$PVC_MAJ_MIN" -eq "$PVC_DEST_MAJ_MIN" ]; then
+		CP_ARGS+=(--no-clobber)
+	else
+		# It's a different disk, we must force copy
+		echo "PVC's source major/minor numbers changed"
+		# --remove-destination will remove each existing destination file before attempting to open it
```

Comment:
Since all this ends up in the pod spec, how about if we move the comments outside the string? There is no need for the pod spec to contain the comments.